src/goto-instrument/dump_c.cpp

```diff
@@ -990,6 +994,72 @@ void dump_ct::convert_global_variable(
 
 /*******************************************************************\
 
+Function: dump_ct::cleanup_harness
+
+Inputs:
+
+Outputs:
+
+Purpose:
+
+\*******************************************************************/
+
+void dump_ct::cleanup_harness(code_blockt &b)
+{
+  replace_symbolt replace;
+  code_blockt decls;
+
+  const symbolt *argc_sym=0;
+  if(!ns.lookup("argc'", argc_sym))
+  {
+    symbol_exprt argc("argc", argc_sym->type);
+    replace.insert(argc_sym->name, argc);
+    code_declt d(argc);
+    decls.add(d);
+  }
+  const symbolt *argv_sym=0;
+  if(!ns.lookup("argv'", argv_sym))
+  {
+    symbol_exprt argv("argv", argv_sym->type);
+    replace.insert(argv_sym->name, argv);
+    code_declt d(argv);
+    decls.add(d);
+  }
+  const symbolt *return_sym=0;
+  if(!ns.lookup("return'", return_sym))
+  {
+    symbol_exprt return_value("return_value", return_sym->type);
+    replace.insert(return_sym->name, return_value);
+    code_declt d(return_value);
+    decls.add(d);
+  }
+
+  Forall_operands(it, b)
+  {
+    codet &code=to_code(*it);
+
+    if(code.get_statement()==ID_function_call)
+    {
+      code_function_callt &call=to_code_function_call(code);
```

Comment:
Same comment as elsewhere about const ref to to_code_function_call(code).function().